metadata-proxy/nginx.conf

```diff
@@ -0,0 +1,63 @@
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+user www-data;
+worker_processes 4;
+pid /run/nginx.pid;
+
+events {
+  worker_connections 20;
+}
+
+http {
+  server {
+    listen 127.0.0.1:988;
```

Comment:
Even if make the port configurable, the addon is still a static file on the master. I think that choosing an unassigned privileged port does more to prevent conflicts than hiding an easy-to-configure setting behind a hard-to-impossible-to-configure implementation.

We can change this if it becomes a problem, but I don't see it being useful until we decide on, implement, and deploy a serious overhaul to the addon manager.